src/main/user-impl/java/com/mysql/cj/jdbc/ha/plugins/failover/FailoverConnectionPlugin.java

```diff
@@ -91,6 +92,25 @@ public class FailoverConnectionPlugin implements IConnectionPlugin {
   static final String METHOD_CLOSE = "close";
   static final String METHOD_IS_CLOSED = "isClosed";
 
+  private static final List<String> METHODS_REQUIRE_UPDATED_TOPOLOGY = new ArrayList<>(Arrays.asList(
+      METHOD_COMMIT,
+      "connect",
+      "isValid",
+      "rollback",
+      "sendQueryCancel",
```

Comment:
Cancelling a query sounds like an emergency action for a user app and I wouldn't do any topology update when it happens.